libraries/core-react/src/components/Checkbox/Input.tsx

```diff
@@ -100,47 +92,49 @@ export const CheckboxInput = forwardRef<HTMLInputElement, InputProps>(
     const fill = disabled
       ? tokens.states.disabled.background
       : tokens.background
-    const { density } = useEds()
 
     const inputWrapperProps = {
-      density,
       disabled,
     }
 
     const inputProps = {
       ref,
       disabled,
       ['data-indeterminate']: indeterminate,
-      density,
       ...rest,
     }
 
+    const { density } = useEds()
+    const token = useToken({ density }, tokens)()
```

Comment:
Suggest moving these to the top as hooks are always declared in the very top of the function as is best practice for clean code when using hooks